compiler/rustc_errors/src/json.rs

```diff
@@ -135,6 +138,25 @@ impl Emitter for JsonEmitter {
         }
     }
 
+    fn emit_timing_section(&mut self, section: TimingSection, kind: TimingSectionKind) {
+        let kind = match kind {
+            TimingSectionKind::Start => "start",
+            TimingSectionKind::End => "end",
+        };
+        let name = match section {
+            TimingSection::Linking => "link",
+        };
+        let timestamp = SystemTime::now()
```

Comment:
I think this should use Instant rather than SystemTime. Using the latter causes issues when changing the system time during a build, for example because your ntp daemon finally got an internet connection.

Reply:
We already use system time for some incr dir names, I think, but you're right that for profiling Instant makes more sense. But then I have to count the duration from some origin time again. Before it was from the creation of the JSON emitter, but that had some issues w.r.t. multiple emitters existing. Should I just take a snapshot when rustc starts or something? :thinking: